src/routes/namespace.go

```diff
@@ -12,70 +11,59 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-func ListNamespaces(client *kubernetes.Clientset, logger *zap.Logger) gin.HandlerFunc {
+func namespaceHandler(logger *zap.Logger, handler func(controller controllers.NamespaceController, c *gin.Context) (interface{}, error)) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		context := c.Request.Context()
-		namespaceController := controllers.New(client, context, logger)
-		namespaces, err := namespaceController.GetNamespaces()
-		if err != nil {
-			c.AbortWithStatusJSON(int(err.(*k8serrors.StatusError).ErrStatus.Code), gin.H{"error": "Failed to list namespaces", "details": err.Error()})
+		client, exists := c.Get("k8sClient")
+		if !exists {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "Kubernetes client not found"})
 			return
 		}
 
-		c.JSON(http.StatusOK, namespaces)
-	}
-}
-
-func GetNamespace(client *kubernetes.Clientset, logger *zap.Logger) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		// TODO: Validate parameter
-		name := c.Param("name")
+		k8sClient := client.(*kubernetes.Clientset)
```

Comment:
I think that we'll need to have it as `kubernetes.Interface` so we can use fake client in our unit tests.

Reply:
Looks like fake client mocks the `clientset` https://pkg.go.dev/k8s.io/client-go/kubernetes/fake.
Also, `kubernetes.Interface` lacks the corev1 functions used in the controller.

Reply:
I didn't work when I added unit tests in my PR. I'll take another look.

Reply:
Nevermind, it's not related to your PR. Sorry.